### PR TITLE
Fixes broken tests and continuous integration

### PR DESCRIPTION
**Pull request by jtfirek**
- fixes all the remaining broken tests (2 have been commented out that @solipsis has to rework posted PEPE)
- Refactors tests to remove duplicate code
- Adds a github actions workflow to assert all tests are passing

### CODE REVIEW

File: test/EigenLayerIntegration.t.sol [hunk 1]
```diff
@@ -100,41 +92,35 @@ contract EigenLayerIntegraitonTest is TestSetup, ProofParsing {
...
-        _upgrade_auction_manager_contract();
-        _upgrade_node_oeprator_manager_contract();
-        _upgrade_etherfi_oracle_contract();
-
         setupRoleRegistry();
     }
+    
+    // Broken by PEPE upgrades
+    // TODO: Update test post PEPE upgrade
+    // function test_fullWithdraw_812() public {
```
Comment (seongyun-ko) [inline test/EigenLayerIntegration.t.sol:100]: is there any way to keep any unit test on `fullWithdraw`?

if you are going to add a new one, we can remove it? @solipsis

Comment (jtfirek) [reply test/EigenLayerIntegration.t.sol:100]: just fyi the `fullWithdraw` function call in this test reverts with the internal  `_distributePayouts` call attempting to distribute more ETH than there is in the Node contract

File: .github/workflows/run-forge-tests.yaml [hunk 2]
```diff
@@ -0,0 +1,30 @@
+name: Run Forge Tests
+
+on:
+  pull_request:
+    branches:
+      - staging-2.5
```
Comment (vvalecha519) [inline .github/workflows/run-forge-tests.yaml:6]: can we do this on all branches

Comment (jtfirek) [reply .github/workflows/run-forge-tests.yaml]: changed to run for all PRs

**Pull request merged**